### PR TITLE
concepts: add courtroom

### PR DESCRIPTION
**Pull request by zelig**
based on @orenyodfat original

### CODE REVIEW

File: concepts/courtroom.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+## Implementation
+
+The courtroom framework include a solidity smart contracts suite, an ABI interface
+specification and tools to interact with these contracts.
+
+A courtroom suite includes  abstract generic `Swear` contract which knows about the tendering and litigation process,
+i.e., how to open a case and conduct a trial.
+An abstract trial is described by a finite state automaton states of which correspond to stages of litigation and
```
Comment (zelig) [inline concepts/courtroom.md]: > a finite state automaton the states of which correspond to...

is it  not ok?

Comment (MrTibbles) [inline concepts/courtroom.md]: by a finite *set of* automation states

Comment (orenyodfat) [inline concepts/courtroom.md]: What about 
"by a finite state automaton which correspond to stages of litigation and..."
https://en.wikipedia.org/wiki/Automaton 
?

Comment (lmars) [inline concepts/courtroom.md]: To clarify, I think the intended meaning was:

```
An abstract trial is described by a finite state
automaton, states of which correspond to...
```

I would prefer the more approachable term "state machine" than "finite state automaton", so:

```
An abstract trial is a state machine which represents
the different stages of litigation...
```

File: concepts/courtroom.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+## Implementation
+
+The courtroom framework includes a solidity smart contracts suite, an ABI interface
+specification and tools to interact with these contracts.
+
+A courtroom suite includes a generic contract which knows how to open a case and conduct a trial.
+An abstract trial is described by a finite state automaton states of which correspond to stages of litigation and
+the transitions are labeled by various outcomes of evaluating evidence submitted to the respective expert witnesses
+associated with the stage.
+The `Swindle` contact just orchestrates the transition through the stages calling the witness contracts, handles grace periods to control the deadline for submitting evidence (challenges or their refutations), reach a guilty/non-guilty verdict accordingly.
```
Comment (MrTibbles) [inline concepts/courtroom.md]: > The `Swindle` contact just orchestrates

The `Swindle` contract orchestrates

**Pull request merged**